### PR TITLE
feat(review): telemetry for review feature usage

### LINKED ISSUE #1065: Review feature emits no telemetry, and callers are indistinguishable
## Problem

The dbt/SQL review feature emits no telemetry. `cli/cmd/review.ts` and `altimate/review/run.ts` contain no `Telemetry` references, and the CLI path creates no session, so it never picks up `session_start` either.

We can't answer basic questions about it: how often it runs, whether it's run by a person or by an agent, whether it finds anything, or whether results get published.

## Also: callers are indistinguishable

`Flag.ALTIMATE_CLI_CLIENT` is already stamped as `source` on every event, but only `cli/cmd/acp.ts` sets it. Every other entry point reports the default `cli`, so a plugin, a CI job and a human look identical.

## Proposal

Two events emitted from a shared helper used by both callers of the review engine — the `review` command and the `dbt_pr_review` tool — distinguished by an `invocation` field:

- `review_run` — one per engine invocation, `status: completed | failed`, plus verdict, mode, tier, and finding counts by severity and category
- `review_post_outcome` — publication result, CLI path only

Caller attribution needs no code change: if the new events don't declare a `source` field, the envelope's existing process-level value passes through, so a caller that exports `ALTIMATE_CLI_CLIENT` is attributed automatically.

Enums, booleans and counts only. Review findings are about customer schema, so no file paths, model or column names, finding text, SQL or diff content.

### PR DESCRIPTION
**Pull request by sahrizvi**
### Issue for this PR

Closes #1065

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor / code improvement
- [ ] Documentation

### What does this PR do?

The review feature emitted no telemetry at all — no `Telemetry` reference in `cli/cmd/review.ts` or the engine, and no session on the CLI path, so not even `session_start`. We couldn't tell how often review runs, who runs it, what it finds, or whether results get published.

Adds two events from one helper shared by **both** callers of the engine — the `review` command and the `dbt_pr_review` tool — separated by an `invocation` field. Instrumenting only the command would have missed every review an agent runs through the tool.

Caller attribution needed no code. `toAppInsightsEnvelopes` already seeds `source` from `Flag.ALTIMATE_CLI_CLIENT`, and an event only overrides that by declaring its own `source` field. These events declare none, so anything exporting `ALTIMATE_CLI_CLIENT` is attributed for free. That let me drop an earlier design carrying a global envelope property, a per-command entrypoint registry, a `reviewPullRequest()` signature change and typed errors through the engine.

Four choices that aren't obvious from the diff:

- **Publication is a separate event.** It happens after the review is computed and can partially succeed, so it can't be a field on `review_run`, and a failed post must not mark a computed review as failed.
- **`partial` covers every degraded post state.** `PostResult` can't distinguish them — `postError` isn't cleared when the retry succeeds, and an inline fallback coexists with a real `reviewId`.
- **Two failure reasons plus a fallback.** The engine degrades rather than throwing for missing manifests, dispatcher failures and the AI lane, so buckets for those would never fire.
- **`by_category` is zero-filled** from the 14-value enum and drops unknown keys, so "never fired" differs from "not possible" and a malformed category can't mint a new dimension.

Enums, booleans and counts only — review findings are about customer schema.

### How did you verify your code works?

- 13 unit tests. Mutation-checked rather than assumed: adding a `source` field to the event fails the attribution test, and removing the zero-fill fails two others.
- One opt-in e2e (`ALTIMATE_E2E=1 bun test test/e2e/`, ~13s) that runs a **real** `altimate-code review` process against a real git repo, with telemetry pointed at a local HTTP sink. Spawned with only `ALTIMATE_CLI_CLIENT` set, the arriving envelope carried `source=plugin:claude-code` — which is the one claim the whole design rests on and can't be shown in-process.
- That run also caught something I'd documented wrongly: a repo with a `dbt_project.yml` but no compiled manifest reports `degraded=true`, so `degraded` is a fidelity flag, not a "no warehouse" flag. Docs corrected.
- Privacy asserted, not assumed: a unit test checks no schema identifier appears in the serialized event, and the e2e greps the whole payload for the fixture's file name, changed SQL and repo path.
- `tsgo --noEmit` clean on `packages/opencode`; 162 telemetry tests passing.

### Screenshots / recordings

N/A — no UI change.

### Checklist

- [x] I have tested my changes locally
- [x] I have not included unrelated changes in this PR

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_018SLUQF3xgZHsGZHSjxe7vb

## Summary by CodeRabbit

* **New Features**
  * Added telemetry for review execution and publication outcomes across CLI and tool-based reviews.
  * Captures completion, failure, duration, verdict, quality metrics, and posting results while protecting repository and customer content.
  * Preserves existing review output and error behavior when telemetry is unavailable.

* **Documentation**
  * Documented review telemetry events, payload fields, privacy safeguards, publication statuses, and event correlation.

### CODE REVIEW

File: docs/docs/reference/telemetry.md [hunk 1]
```diff
@@ -47,6 +47,19 @@ We collect the following categories of events:
...
 | `validator_retries_exhausted` | A session terminated with unresolved validator failures after exhausting the synthetic-retry budget — names of the failing validators (no failure body content). |
+| `review_run` | A dbt/SQL review completed or failed — `invocation` (`cli` for `altimate-code review`, `tool` for the `dbt_pr_review` tool), status, duration, and on success the verdict, the pre-gating verdict, mode, risk tier, and finding counts by severity and by category. No file paths, model or column names, finding titles or bodies, SQL, diff content, or repository/branch/PR names. |
+| `review_post_outcome` | Whether a review was published to GitHub — `not_requested`, `not_attempted`, `target_unresolved`, `full`, `partial`, or `summary_failed`, plus duration. Exactly one per **completed** review: a review that failed emits `review_run: failed` and no post event, so absence means the review failed rather than that an event was lost. `not_attempted` is publication requested but never reached (a bad `--output` path, a stdout write error). No repository, PR, or comment content. |
```
[the remark this answers is not in the document]
Comment (sahrizvi) [reply docs/docs/reference/telemetry.md]: Fixed in `67a7729`. Correct, and it was a real error in wording I added last commit — the `dbt_pr_review` tool completes reviews and never publishes, so as written an analyst correlating events would have read every `invocation: tool` completed review as a failure. The row now scopes the cardinality to the CLI path and calls the tool path out explicitly.

File: packages/opencode/test/altimate/review/telemetry.test.ts [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+      bodies.push(String(init?.body ?? ""))
+      return new Response("", { status: 200 })
+    }) as unknown as typeof fetch)
+
+    try {
+      delete process.env.ALTIMATE_TELEMETRY_DISABLED
+      process.env.APPLICATIONINSIGHTS_CONNECTION_STRING =
+        "InstrumentationKey=k;IngestionEndpoint=https://example.invalid"
+      process.env.ALTIMATE_CLI_CLIENT = "plugin:claude-code"
+      await Telemetry.init()
```
[the remark this answers is not in the document]
Comment (sahrizvi) [reply packages/opencode/test/altimate/review/telemetry.test.ts:291]: Fixed in `67a7729`, as suggested. Confirmed rather than assumed: adding a prior `Telemetry.init()` to this same file makes the test fail without the `shutdown()` and pass with it. Also restored mocks at the top of the test — before installing its own fetch spy, since restoring afterwards removes that spy — because sibling describes here spy `Telemetry.track` and this is the only test needing the real one. Added `expect(Telemetry.isEnabled()).toBe(true)` so a future regression fails with a cause instead of on an empty batch, and switched `bodies[0]` to a flatMap over all bodies. The identical bug was just fixed on the sibling onboarding PR, where it had been failing CI.

File: packages/opencode/test/altimate/review/telemetry.test.ts [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+      emitReviewRun({ invocation: "cli", durationMs: 1, sessionID: "", envelope: envelope() })
+      emitReviewPostOutcome({ outcome: "not_requested", durationMs: 0, sessionID: "" })
+      await Telemetry.flush()
```
[the remark this answers is not in the document]
Comment (sahrizvi) [reply packages/opencode/test/altimate/review/telemetry.test.ts:298]: Fixed in `67a7729` — same conclusion `cubic` reached independently, and the fix is your diff. Verified by adding a prior `init()` to this file: fails without the `shutdown()`, passes with it. Also hardened against a surviving `Telemetry.track` spy from a sibling describe and added an `isEnabled()` assertion so the next failure of this shape reports its cause.

File: packages/opencode/test/altimate/review/telemetry.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sahrizvi) [reply packages/opencode/test/altimate/review/telemetry.test.ts:291]: Fixed in `67a7729`. `HOME`/`USERPROFILE` now point at a temp dir, restored and removed in `finally` — the pattern the e2e in this PR already used, as you note. Real `init()` is intentional here (the attribution claim is only observable after serialization); minting a machine-id in the developer's real home was not.

File: docs/docs/reference/telemetry.md [hunk 4]
```diff
@@ -47,6 +47,19 @@ We collect the following categories of events:
...
+
+### Notes on the review events
```
[the remark this answers is not in the document]
Comment (sahrizvi) [reply docs/docs/reference/telemetry.md:70]: Fixed in `67a7729` — the general sentence is now hoisted above the `### Notes on the review events` subsection, so it reads as applying to all events again.

File: docs/docs/reference/telemetry.md [hunk 5]
```diff
@@ -47,6 +47,19 @@ We collect the following categories of events:
...
+
+- `degraded` is a fidelity flag, not a warehouse flag. It is set when a review found no reviewable
+  files, had no usable manifest for the changed models, or surfaced a finding whose analysis was
+  undecidable. It does not mean "no warehouse was connected".
+- The category breakdown counts findings that were actually surfaced — after de-duplication, rubric
+  exclusion, and the severity threshold. It is not a count of raw rule detections, and it is grouped
+  by category rather than by individual rule.
+- Reviews run through the `dbt_pr_review` tool also emit the standard `tool_call` event. They are
+  the same review; count `review_run` rather than both.
```
[the remark this answers is not in the document]
Comment (sahrizvi) [reply docs/docs/reference/telemetry.md:79]: Fixed in `67a7729`, same as `cubic` flagged. Took the second option: moved the "Each event includes a timestamp…" sentence above the `### Notes on the review events` heading rather than moving the subsection down.

**Review by saravmajestic — DISMISSED** [review]

**Review by dev-punia-altimate — APPROVED** [review]
approving as requested
